### PR TITLE
do now allow duplicated items for resources blocked

### PR DESCRIPTION
**Pull request by cezaraugusto**
fix https://github.com/brave/brave-browser/issues/420

Test Plan:

covered by automated tests

### CODE REVIEW

File: app/helpers/arrayUtils.ts [hunk 1]
```diff
@@ -0,0 +1,5 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+export const unique = (arr: Array<string>) => [...new Set(arr)]
```
Comment (bbondy) [inline app/helpers/arrayUtils.ts]: nice way to deduplicate items :)

File: test/app/components/braveShields/braveShieldsStatsTest.tsx [hunk 2]
```diff
@@ -96,6 +96,26 @@ describe('BraveShieldsStats component', () => {
...
       assert.equal(assertion2, true)
     })
+
+    it('properly merges ads + trackers in the same list without duplicates', () => {
+      const componentId = 'totalAdsTrackersBlockedStat'
+      const target = { target: { id: componentId } }
+      const newProps = Object.assign(fakeProps, {
+        adsBlockedResources: [
+          'https://everbodys-got-something-to-hide-except-me-and-my-monkey.cool',
+          'https://everbodys-got-something-to-hide-except-me-and-my-monkey.cool'
```
Comment (bbondy) [inline test/app/components/braveShields/braveShieldsStatsTest.tsx]: 🐒

File: app/state/shieldsPanelState.ts [hunk 3]
```diff
@@ -74,22 +75,22 @@ export const updateResourceBlocked: shieldState.UpdateResourceBlocked = (state,
 
   if (blockType === 'ads') {
     tabs[tabId].adsBlocked++
-    tabs[tabId].adsBlockedResources = [ ...tabs[tabId].adsBlockedResources, subresource ]
+    tabs[tabId].adsBlockedResources = unique([ ...tabs[tabId].adsBlockedResources, subresource ])
```
Comment (yrliou) [inline app/state/shieldsPanelState.ts]: Our block counts will not match the counts of resources we are showing, I think it might be better to match both.

Comment (bbondy) [reply app/state/shieldsPanelState.ts]: `Maybe set adsBlocked` to the length of adsBlockedResources after this line?

**Comment by cezaraugusto** [comment]
ok ready w/ updates:

* c8edd0f82f8dcad59e1475907aeea981c5794eea was added as a requirement after this PR review. This addresses https://github.com/brave/brave-extension/pull/42#pullrequestreview-134091073 as a follow-up /cc @yrliou 

Note: an ad that is also blocked as tracker is kept as-is and **not deduped**. Visiting https://cezaraugusto.net will show 2 resources blocked with the same analytics url. Workaround for this took a lot of code changes and not worth the effort. Better take would be just making clear for adblock what/what not is an ad/tracker.

**Pull request merged**